### PR TITLE
Revert "backupccl: protect entire keyspan during cluster backup"

### PR DESCRIPTION
**Pull request by adityamaru**
This reverts commit 1b5fd4f000cfd8e5085b0b67e6db0d9fa5feb69b.

The commit above laid a pts record over the entire table keyspace.
This did not account for two things (with the potential of there being
more):

1. System tables that we do not backup could have a short GC TTL, and
so incremental backups that attempt to protect from `StartTime` of the
previous backup would fail.

2. Dropped tables often have a short GC TTL to clear data once they have
been dropped. This change would also attempt to protect "dropped but not
gc'ed tables" even though we exclude them from the backup, and fail on
pts verification.

One suggested approach is to exclude all objects we do not backup by
subtracting these spans from {TableDataMin, TableDataMax}. This works
for system tables, and dropped but not gc'ed tables, but breaks for
dropped and gc'ed tables. A pts verification would still find the leaseholder
of the empty span and attempt to protect below the gc threshold.

In conclusion, we need to think about the semantics a little more before
we rush to protect a single key span.

### CODE REVIEW

**Review by dt — APPROVED** [review]

**Comment by adityamaru** [comment]
bors r=dt
